chia/data_layer/data_store.py

```diff
@@ -1073,6 +1102,77 @@ async def delete(
 
         return new_root
 
+    async def upsert(
```

Comment:
Do we need side and reference hash control for the insert case of upsert?

Reply:
I think it should be equivalent to an autoinsert, otherwise we'll just complicate the interface (when proper use should be `insert` anyway...)